StreamingASR/StreamingASR/app/src/main/cpp/MainActivityJNI.cpp

```diff
@@ -0,0 +1,92 @@
+#include <jni.h>
+#include <string>
+#include <Eigen/Dense>
+#include "librosa/librosa.h"
+#include <iostream>
+#include <vector>
+#include <chrono>
+#include <numeric>
+#include <algorithm>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+
+using namespace std;
+
+using Eigen::MatrixXd;
+
+extern "C" JNIEXPORT jobject JNICALL
+Java_org_pytorch_demo_streamingasr_MainActivity_melSpectrogram(JNIEnv* env, jobject obj,
+                                                                    jdoubleArray data) {
+    int len = env -> GetArrayLength(data);
+    std::vector<float> x;
+    jdouble *element = env->GetDoubleArrayElements(data, 0);
+    for(int i=0; i<len; ++i) {
+        x.push_back((float)element[i]);
+    }
+
+    int n_fft = 400;
+    int n_hop = 160;
+    int n_mel = 80;
+    int fmin = 0;
+    int fmax = 8000;
+    int sr = 16000;
+
+    std::vector<std::vector<std::complex<float>>> X = librosa::Feature::stft(x, n_fft, n_hop, "hann", true, "reflect");
+
+    std::vector<std::vector<float>> mels = librosa::Feature::melspectrogram(x, sr, n_fft, n_hop, "hann", true, "reflect", 2.f, n_mel, fmin, fmax);
+
+    jclass vectorClass = env->FindClass("java/util/Vector");
```

Comment:
Can we use `java.util.ArrayList` instead or do we really need synchronized Vector?

Reply:
Maybe we can.. but guess it's a nice to have?